lib/core/Core.ts

```diff
@@ -102,41 +101,31 @@ export default class Core {
 
     if (this.config.observingIntervalInSeconds > 0) {
       await this.observer.startPeriodicProcessing();
+      // Only pull blockchain time when observer is enabled.
+      await this.blockchainClock.startPeriodicPullLatestBlockchainTime();
     } else {
       Logger.warn(LogColor.yellow(`Transaction observer is disabled.`));
     }
 
+    // Performance optimization to cache the time so requests don't have to go to db
+    this.blockchainClock.startPeriodicCacheTime();
```

Comment:
What if we make it more generic by doing something like this:

```typescript
const enableRealBlockchainTimePull = this.config.observingIntervalInSeconds > 0;
this.blockchainClock.initialize(enableRealBlockchainTimePull);
```

With this we can then further simplify the two `setTimeout()` inside `BlockchainClock` into just one, I don't see them need to be run on different schedule.